### ISSUE TITLE
Analysis files from wrong samples removed

### ISSUE THREAD

**Issue by parlundin** [NationalGenomicsInfrastructure/ngi_pipeline#298]
When restarting NGI_PIPELINE for a sample where previous analysis already been performed, the previous analysis files are removed from the analysis folders 01 to 07, unfortunately it looks like it in some cases removes the files for other samples as well.

**Comment by Galithil** [comment]
I pushed a bugfix for that, but I'm going to need to test it. Is there a way to try it without doing that in production ? reproducing the bug seems a little hard

**Comment by mariogiov** [comment]
Yeah this is weird. @parlundin do you have an example you can direct us to?

**Comment by Galithil** [comment]
patrik's words : 
>I believe it happend when I restarted a failed sample and chosed to remove the old files. Lets say I restarted sample 10, it then would remove all samples with 10 in the sample name.

I also found out that the sub handling the "which files to copy/delete" does so without knowing which sample to work on, it only has project-level information, so I was wondering if that could have an impact

**Comment by mariogiov** [comment]
The NGIProject object that is passed to the function as an argument has as children NGISample objects, and the removal function just parses through these. But the trouble is here:
https://github.com/NationalGenomicsInfrastructure/ngi_pipeline/blob/devel/ngi_pipeline/engines/piper_ngi/utils.py#L171
It's a naive `glob` that I guess assumed more unique sample names. This should be an easy fix though.

**Comment by Galithil** [comment]
yep, just did that. Now to hope that this still works...
